pkg/asset/agent/image/cpio.go

```diff
@@ -91,9 +92,26 @@ func (ca *CpioArchive) StoreFile(filename string, dstPath string) error {
 		return err
 	}
 
-	_, err = io.Copy(ca.cpioWriter, f)
-	if err != nil {
-		return err
+	fm := fileInfo.Mode()
+	switch {
+	case fm.IsRegular():
+		// Copy the file content
```

Comment:
nit: could move the `Open()` call to here.

Reply:
Thanks, good point